spikingjelly/activation_based/triton_kernel/torch2triton/torch2graph.py

```
@@ -115,21 +144,31 @@ def generate_forward_and_backward_graph(
         partition_fn=min_cut_rematerialization_partition,
     )
 
+    # Build local inputs so we don't mutate callers' tensors. Set requires_grad
+    # on detached clones as needed to avoid modifying non-leaf tensors.
+    local_inputs = []
     if requires_grad is not None:
         if len(requires_grad) != len(example_inputs):
             raise ValueError(
                 "requires_grad must have the same length as example_inputs"
             )
         for i, r in zip(example_inputs, requires_grad):
```

Ruff (0.15.13)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)